support-workers/src/main/scala/com/gu/zuora/subscriptionBuilders/DigitalSubscriptionBuilder.scala

```diff
@@ -82,15 +83,20 @@ object DigitalSubscriptionBuilder {
       initialTerm = initialTerm
     )
 
-    val withRedemptionCode = generateRedemptionCode(readerType, purchase.billingPeriod, subscriptionData)
-    val withPromoApplied = applyPromoCode(purchase.promotionService, purchase.maybePromoCode, purchase.country, productRatePlanId, withRedemptionCode)
+    val withRedemptionCode = addRedemptionCodeIfGift(readerType, purchase.billingPeriod, subscriptionData)
+    val withPromoApplied = applyPromoCodeIfPresent(purchase.promotionService, purchase.maybePromoCode, purchase.country, productRatePlanId, withRedemptionCode)
 
     EitherT.fromEither[Future](withPromoApplied.left.map(Left.apply))
   }
 
-  def generateRedemptionCode(readerType: ReaderType, billingPeriod: BillingPeriod, subscriptionData: SubscriptionData) = {
+  def fromBillingPeriod(billingPeriod: BillingPeriod): GiftDuration = billingPeriod match {
```

Comment:
flip the name now it's here
```suggestion
  def toGiftDuration(billingPeriod: BillingPeriod): GiftDuration = billingPeriod match {
```